crates/db_schema/src/utils.rs

```diff
@@ -345,10 +349,34 @@ pub fn diesel_url_create(opt: Option<&str>) -> LemmyResult<Option<DbUrl>> {
   }
 }
 
+/// Sets a few additional config options necessary for starting lemmy
+fn build_config_options_uri_segment(config: &str) -> String {
+  let mut url = Url::parse(config).expect("Couldn't parse postgres connection URI");
+
+  // Set `lemmy.protocol_and_hostname` so triggers can use it
+  let lemmy_protocol_and_hostname_option =
+    "lemmy.protocol_and_hostname=".to_owned() + &SETTINGS.get_protocol_and_hostname();
+  let mut options = CONNECTION_OPTIONS.to_vec();
+  options.push(&lemmy_protocol_and_hostname_option);
+
+  // Create the connection uri portion
+  let options_segments = options
+    .iter()
+    .map(|o| "-c ".to_owned() + o)
+    .collect::<Vec<String>>()
+    .join(" ");
+
+  url.set_query(Some(&format!("options={options_segments}")));
```

Comment:
This way is easier: https://docs.rs/url/latest/url/struct.Url.html#method.query_pairs_mut

Reply:
I tried this one, but had to revert it since it overwrites some equals that are necessary for inside the options segments: https://stackoverflow.com/a/42131826/1655478

Reply:
Would be good to write a quick unit test for this function.

Reply:
Its not really too necessary because every single DB command will fail if this one's wrong. Even if its missing the `lemmy.procotol_and_hostname` config option.